lib/backend/sqlbk/config.go

```diff
@@ -127,14 +142,36 @@ func (c *Config) CheckAndSetDefaults() error {
 	if c.Addr == "" {
 		return trace.BadParameter("Addr is required")
 	}
-	if c.TLS.CAFile == "" {
-		return trace.BadParameter("TLS.CAFile is required")
-	}
-	if c.TLS.ClientKeyFile == "" {
-		return trace.BadParameter("TLS.ClientKeyFile is required")
-	}
-	if c.TLS.ClientCertFile == "" {
-		return trace.BadParameter("TLS.ClientCertFile is required")
+
+	// TODO: find a way to have less boilerplate here
```

Comment:
I think without resorting to some sort of field-tag based validation system, we are always going to be stuck with these verbose (but at least easily coverage tested) if blocks for validation.

As this `CheckAndSetDefaults` grows a bit more complex, it would certainly be nice to have a super simple unit test for it.

Reply:
Added a test with a handful of test cases for `Config.CheckAndSetDefaults`, which also made me realize that we're going to have to redo the configuration interface if we add a different driver or another authentication mechanism 😬

Reply:
😨 

Well thanks for adding the tests anywhoo ! I've gone ahead and approved this now.

Reply:
Well, we might also get lucky and all the alternate SQL drivers will support all the authentication mechanisms we add 😅